### PR TITLE
Update from go-client 1.10 to 1.17.3

### PR DESCRIPTION
**Pull request by sdake**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,10 +3,9 @@ module istio.io/tools
 go 1.12
 
 replace (
-	k8s.io/api => k8s.io/api v0.0.0-20190817221950-ebce17126a01
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.0.0-20191011152811-a1d7614a8e0f
-	k8s.io/apimachinery => k8s.io/apimachinery v0.0.0-20190817221809-bf4de9df677c
-	k8s.io/client-go => k8s.io/client-go v10.0.0+incompatible
+	k8s.io/api => k8s.io/api v0.17.3
```
Comment (howardjohn) [inline go.mod]: replace no longer needed now, they fixed their go.mod issues

File: go.mod [hunk 2]
```diff
@@ -23,33 +22,25 @@ require (
...
-	gopkg.in/inf.v0 v0.9.1 // indirect
-	gopkg.in/yaml.v2 v2.2.4
+	gopkg.in/yaml.v2 v2.2.8
 	istio.io/gogo-genproto v0.0.0-20191009201739-17d570f95998
-	k8s.io/api v0.0.0-20191004120003-3a12735a829a
+	k8s.io/api v0.17.3
 	k8s.io/apiextensions-apiserver v0.0.0-20191011152811-a1d7614a8e0f
-	k8s.io/apimachinery v0.0.0-20191004115801-a2eda9f80ab8
+	k8s.io/apimachinery v0.17.3
 	k8s.io/client-go v0.0.0-20191016111102-bec269661e48
```
Comment (howardjohn) [inline go.mod]: update this as well?

**Comment by sdake** [comment]
@howardjohn This is, I hope, a solution to the build-tools container build busted in test-infra.

I'd like to match what is in test-infra first (which is 0.17.3 - where I think the regression in the build-tools started). I am happy to submit a quick followup after I backport this to release-1.5 for master for 1.18.

Cheers
-steve

File: go.mod (hunk 2 above)
Comment (sdake) [reply go.mod]: thanks. resolved.

**Comment by howardjohn** [comment]
We should not be depending on our go.mod file when fetching dependencies or we will continue to run into issues. Maybe we need to use something like https://github.com/myitcv/gobin - not sure if there are other solutions

**Comment by sdake** [comment]
Agreed. cool tool - I will give it a try in the build-tools container.

Cheers
-steve

**Comment by sdake** [comment]
closing PR in favor of https://github.com/istio/tools/pull/855.

We probably need to take a look at all of our deps in our various ancillary repos and bring them up to date.

**Pull request closed without merging**